tutorials/ComplexArithmetic/Workbook_ComplexArithmetic.ipynb

```diff
@@ -0,0 +1,953 @@
+    "\n",
+    "|Power of I|$i^1$|$i^2$|$i^3$|$i^4$|$i^5$|$i^6$|$i^7$|$i^8$|$\\dots$|\n",
+    "|----|----|----|----|----|----|----|----|----|----|\n",
+    "|Result|$i$|$-1$|$i$|$1$|$i$|$-1$|$i$|$1$|$\\dots$|\n",
+    "\n",
+    "This means that $i^{(n \\cdot 4)}$ will always give $1$ if $n$ is a whole number.\n",
+    "\n",
+    "*Python note*:  \n",
+    ">`%` is the python modulo operator it returns the remainder of a division. For example: `7%2` gives 1, because 1 remains after $\\frac{7}{3}$, another example would be `10%5` which returns 0.  \n",
+    ">We can use this to know if $n$ in $i^n$, could be wholly divided by 4 for which we will use in the exercise.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Success!\n"
+     ]
+    }
+   ],
+   "source": [
+    "@exercise\n",
+    "def imaginary_power(n : int) -> int:\n",
+    "    # If n is divisible by 4\n",
+    "    if n % 4 == 0:\n",
+    "        return 1\n",
+    "    else:\n",
+    "        return -1"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "[Return to task 1 of the Complex Arithmetic kata.](./ComplexArithmetic.ipynb#Exercise-1:-Powers-of-$i$.)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Complex Numbers\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### <span style=\"color:blue\">Exercise 2</span>: Complex addition.\n",
+    "\n",
+    "**Inputs:**\n",
+    "\n",
+    "1. A complex number $x = a + bi$, represented as a tuple `(a, b)`.\n",
+    "2. A complex number $y = c + di$, represented as a tuple `(c, d)`.\n",
+    "\n",
+    "**Goal:** Return the sum of these two numbers $x + y = z = g + hi$, represented as a tuple `(g, h)`.\n",
+    "\n",
+    "> A tuple is a pair of numbers.\n",
+    "> You can make a tuple by putting two numbers in parentheses like this: `(3, 4)`.\n",
+    "> * You can access the $n$th element of tuple `x` like so: `x[n]`\n",
+    "> * For this tutorial, complex numbers are represented as tuples where the first element is the real part, and the second element is the real coefficient of the imaginary part\n",
+    "> * For example, $1 + 2i$ would be represented by a tuple `(1, 2)`, and $7 - 5i$ would be represented by `(7, -5)`.\n",
+    ">\n",
+    "> You can find more details about Python's tuple data type in the [official documentation](https://docs.python.org/3/library/stdtypes.html#tuples).\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Solution\n",
+    "\n",
+    "Adding two complex numbers can be done by separately adding the real part of the number and the imaginary part.  \n",
+    "$ x = a + bi $  \n",
```

Comment:
I wouldn't repeat the definitions for x and y, since they've just been mentioned in the task statement a few lines before. The longer the solution is, the scarier it looks, so I try not to duplicate things unnecessarily.  (Same comment for the later tasks)